src/components/icon/icon-state.broadcast.ts

```diff
@@ -0,0 +1,89 @@
+import { internalIcons } from './internal-icons-lib.js';
+import { SvgIconParser } from './registry/parser.js';
+import type {
+  BroadcastIconsChangeMessage,
+  Collection,
+  IconMeta,
+  SvgIcon,
+} from './registry/types.js';
+import { ActionType } from './registry/types.js';
+
+export class IconsStateBroadcast {
+  private parser: SvgIconParser;
+  private iconBroadcastChannel: BroadcastChannel;
+  constructor(
+    collections: Collection<string, Map<string, SvgIcon>>,
+    refsCollection: Collection<string, Map<string, IconMeta>>
+  ) {
+    this.parser = new SvgIconParser();
+    this.iconBroadcastChannel = new BroadcastChannel('ignite-ui-icon-channel');
+    this.iconBroadcastChannel.onmessage = (event) => {
```

Comment:
Nitpick: I would prefer this to be `addEventListener('message', handler)`, and make the handler a separate class method.